lib/SILGen/SILGenConvert.cpp

```diff
@@ -654,6 +654,40 @@ ManagedValue SILGenFunction::emitExistentialErasure(
     return emitManagedRValueWithCleanup(existential);
   }
   case ExistentialRepresentation::Opaque: {
+  
+    // If the concrete value is a pseudogeneric archetype, first erase it to
+    // its upper bound.
+    auto anyObjectProto = getASTContext()
+      .getProtocol(KnownProtocolKind::AnyObject);
+    auto anyObjectTy = anyObjectProto
+      ? anyObjectProto->getDeclaredType()->getCanonicalType()
```

Comment:
It seems like it would be more robust to define an AnyObject protocol in the test and remove this hack.

Do you mind doing that at some point on master, or just filing a JIRA?